integration/networking/bridge_test.go

```diff
@@ -0,0 +1,350 @@
+			linkLocal: true,
+		},
+		{
+			name:    "IPv6 link-local address on internal network",
+			pingCmd: []string{"ping", "-c1", "-W3", "-6"},
+			bridgeOpts: []func(create *types.NetworkCreate){
+				network.WithIPv6(),
+				network.WithInternal(),
+				// See the note above about link-local addresses.
+				network.WithIPAM("fe80::/64", "fe80::1"),
+			},
+			linkLocal: true,
+			skipMsg:   "See moby/moby#45649",
+		},
+	}
+
+	for tcID, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			skip.If(t, tc.skipMsg != "", tc.skipMsg)
+			ctx := context.Background()
+
+			bridgeName := fmt.Sprintf("testnet-icc-%d", tcID)
+			network.CreateNoError(ctx, t, c, bridgeName, append(tc.bridgeOpts,
+				network.WithDriver("bridge"),
+				network.WithOption("com.docker.network.bridge.name", bridgeName))...)
+			defer network.RemoveNoError(ctx, t, c, bridgeName)
+
+			ctr1Name := fmt.Sprintf("ctr-icc-%d-1", tcID)
+			id1 := container.Run(ctx, t, c,
+				container.WithName(ctr1Name),
+				container.WithImage("busybox:latest"),
+				container.WithCmd("/bin/sleep", "infinity"),
+				container.WithNetworkMode(bridgeName))
+			defer c.ContainerRemove(ctx, id1, types.ContainerRemoveOptions{
+				Force: true,
+			})
+
+			pingHost := ctr1Name
+			if tc.linkLocal {
+				inspect := container.Inspect(ctx, t, c, id1)
+				pingHost = inspect.NetworkSettings.Networks[bridgeName].GlobalIPv6Address + "%eth0"
+			}
+
+			ctr2Name := fmt.Sprintf("ctr-icc-%d-2", tcID)
+			id2 := container.Run(ctx, t, c,
+				container.WithName(ctr2Name),
+				container.WithImage("busybox:latest"),
+				container.WithCmd(append(tc.pingCmd, pingHost)...),
+				container.WithNetworkMode(bridgeName))
+			defer c.ContainerRemove(ctx, id2, types.ContainerRemoveOptions{
+				Force: true,
+			})
+
+			exitCode, err := d.ContainerExitCode(t, id2)
+			assert.NilError(t, err)
+			assert.Check(t, exitCode == int64(0))
+
+			logs := getContainerLogs(t, ctx, c, id2)
+			assert.Check(t, strings.Contains(logs, "1 packets transmitted, 1 packets received"))
+
+			if t.Failed() {
+				t.Logf("Logs from %s:\n%s", ctr2Name, logs)
+			}
+		})
+	}
+}
+
+// TestBridgeIPv6SLAACLLAddress is pretty much the same as TestBridgeICC, but this one checks whether SLAAC addresses
+// assigned by the kernel work properly. To do so, the MAC address of ctr1 is manually assigned.
+func TestBridgeIPv6SLAACLLAddress(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+
+	d := daemon.New(t)
+	d.StartWithBusybox(t, "-D", "--experimental", "--ip6tables")
+	defer d.Stop(t)
+
+	c := d.NewClientT(t)
+	defer c.Close()
+
+	testcases := []struct {
+		name       string
+		bridgeOpts []func(create *types.NetworkCreate)
+		skipMsg    string
+	}{
+		{
+			name:       "IPv6 non-internal network",
+			bridgeOpts: []func(create *types.NetworkCreate){},
+		},
+		{
+			name: "IPv6 internal network",
+			bridgeOpts: []func(create *types.NetworkCreate){
+				network.WithInternal(),
+			},
+			skipMsg: "See moby/moby#45649",
```

Comment:
nit: maybe use the actual URL, so that it's clickable for those that want to view it.

- This looks to link to a merged pull request; do we have more details? https://github.com/moby/moby/pull/45649